# Practice_Challenge_1 Tasks

#### PART A

In this project, we would like to create a portfolio of 3 cryptocurrencies ( BTC, ETH, and ADA ) and 3 stocks ( AMZN, TSLA, and AAPL).

Using Pandas perform the following:
===================================

1. Using Yahoo Finance, search for the current closing price of the aforementioned list of cryptos and stocks

2. create a Python dictionary that lists the ticker symbol(Keys) and prices (values) 

3. create a pandas series from a dictionary to show the cryptos/stocks ticker symbols and their corresponding prices

4. Assume that you have the following number of assets in your portfolio: BTC = 3, ETH = 10, ADA = 1000, AMZN = 20, TSLA = 20, APPL = 5.
Create a new Pandas series that contains these number of securities. Use the ticker symbols as the Pandas series lables.

5. Multiply the number of securities by the price of each security. Calculate the total value of the portfolio including all securities


In [1]:
import pandas as pd

#### Task 1 & 2

1. Using Yahoo Finance, search for the current closing price of the aforementioned list of cryptos and stocks

2. create a Python dictionary that lists the ticker symbol(Keys) and prices (values) 

In [2]:
portfolio = {'BTC':  43500,
             'ETH':  3200,
             'ADA':  1,
             'AAPL': 171,
             'TSLA': 1000, 
             'AMZN': 3175}

In [3]:
portfolio

{'BTC': 43500, 'ETH': 3200, 'ADA': 1, 'AAPL': 171, 'TSLA': 1000, 'AMZN': 3175}

#### Task 3

3. create a pandas series from a dictionary to show the cryptos/stocks ticker symbols and their corresponding prices


In [7]:
portfolio_series = pd.Series( data = portfolio)
portfolio_series

BTC     43500
ETH      3200
ADA         1
AAPL      171
TSLA     1000
AMZN     3175
dtype: int64

In [8]:
type(portfolio_series)

pandas.core.series.Series

In [9]:
portfolio_series.squeeze()

BTC     43500
ETH      3200
ADA         1
AAPL      171
TSLA     1000
AMZN     3175
dtype: int64

In [10]:
type(portfolio_series)

pandas.core.series.Series

#### Task 4

In [ ]:
4. Assume that you have the following number of assets in your portfolio: BTC = 3, ETH = 10, ADA = 1000, AMZN = 20, TSLA = 20, APPL = 5.
Create a new Pandas series that contains these number of securities. 

Use the ticker symbols as the Pandas series lables.

In [11]:
number_securities = pd.Series(data = [10, 5, 2, 4, 5, 6],  index = ['BTC', 'ETH', 'ADA', 'AAPL', 'TSLA', 'AMZN'])
number_securities

BTC     10
ETH      5
ADA      2
AAPL     4
TSLA     5
AMZN     6
dtype: int64

#### Task 5

5. Multiply the number of securities by the price of each security. Calculate the total value of the portfolio including all securities

In [16]:
Total_dollar_value = portfolio_series.mul(number_securities)
Total_dollar_value

BTC     435000
ETH      16000
ADA          2
AAPL       684
TSLA      5000
AMZN     19050
dtype: int64

In [18]:
print('Total portfolio Value = ${}'.format(Total_dollar_value.sum()))

Total portfolio Value = $475736


In [19]:
print(f'TOtal portfolio Value = ${Total_dollar_value.sum()}')

TOtal portfolio Value = $475736


## PART - B

Using the attached "S&P500_Proces.csv" perform the following:

1. Using pandas, read the csv file into Pandas Data Frame. Check the datatype.

2. Using pandas, read the csv fiel into a Pandas Data Series. Check the datatype.

3. Obtain the maximum, minimum and average values of the S&P500

4. Sort the Pandas Series in an ascending and descending order. Enforce the change in memory.

5. Use describe() method to obtain a statistical summary of S&P500. What is standard deviation?

6. With and without rounding the series to the nearest integer, search the price 3349

#### Task 1

In [23]:
#1. Using pandas, read the csv file into Pandas Data Frame. Check the datatype.

SP_prices = pd.read_csv('S&P500_Prices.csv')
SP_prices

sp500
0     1295.500000
1     1289.089966
2     1293.670044
3     1308.040039
4     1314.500000
...           ...
2154  3327.770020
2155  3349.159912
2156  3351.280029
2157  3360.469971
2158  3333.689941

[2159 rows x 1 columns]

In [24]:
type(SP_prices)

pandas.core.frame.DataFrame

#### Task 2

2. Using pandas, read the csv fiel into a Pandas Data Series. Check the datatype.


In [32]:
SP_prices_series = SP_prices.squeeze()

In [33]:
type(SP_prices_series)

pandas.core.series.Series

#### Task 3

In [ ]:
3. Obtain the maximum, minimum and average values of the S&P500


In [40]:
print(SP_prices_series.max())

3386.149902


In [41]:
print(SP_prices_series.min())

1278.040039


In [50]:
# find avg - approach 1

print(SP_prices_series.mean())

2218.7495540592868


In [49]:
# find avg - approach 2

print(SP_prices_series.sum() / SP_prices_series.count())

2218.7495540592868


#### Task 4

4. Sort the Pandas Series in an ascending and descending order. Enforce the change in memory.


In [43]:
SP_prices_series.sort_values(inplace = True)
SP_prices_series

97      1278.040039
98      1278.180054
99      1285.500000
1       1289.089966
2       1293.670044
           ...     
2038    3373.229980
2034    3373.939941
2033    3379.449951
2035    3380.159912
2037    3386.149902
Name: sp500, Length: 2159, dtype: float64

In [44]:
SP_prices_series.sort_index(inplace = True)
SP_prices_series

0       1295.500000
1       1289.089966
2       1293.670044
3       1308.040039
4       1314.500000
           ...     
2154    3327.770020
2155    3349.159912
2156    3351.280029
2157    3360.469971
2158    3333.689941
Name: sp500, Length: 2159, dtype: float64

In [45]:
SP_prices_series.sort_index(ascending = False, inplace = True)
SP_prices_series

2158    3333.689941
2157    3360.469971
2156    3351.280029
2155    3349.159912
2154    3327.770020
           ...     
4       1314.500000
3       1308.040039
2       1293.670044
1       1289.089966
0       1295.500000
Name: sp500, Length: 2159, dtype: float64

#### Task 5

5. Use describe() method to obtain a statistical summary of S&P500. What is standard deviation?


In [46]:
SP_prices_series.describe()

count    2159.000000
mean     2218.749554
std       537.321727
min      1278.040039
25%      1847.984985
50%      2106.629883
75%      2705.810059
max      3386.149902
Name: sp500, dtype: float64

Standard deciation is 537.321727

#### Task 6

6. With and without rounding the series to the nearest integer, search the price 3349

In [52]:
SP_prices_series

2158    3333.689941
2157    3360.469971
2156    3351.280029
2155    3349.159912
2154    3327.770020
           ...     
4       1314.500000
3       1308.040039
2       1293.670044
1       1289.089966
0       1295.500000
Name: sp500, Length: 2159, dtype: float64

In [54]:
3349 in SP_prices_series.values

False

In [55]:
3349.159912 in SP_prices_series.values

True

In [62]:
BTC_price = pd.read_csv('crypto.csv')
BTC_price.squeeze()
type(BTC_price)

pandas.core.frame.DataFrame